### PR TITLE
VP-1975 Fix: add validation to folder create operation

### PR DESCRIPTION
**Pull request by Vectorfield4**
https://virtocommerce.atlassian.net/browse/VP-1975

### Proposed of changes
Describe the technical details of realization provided by you.

### Make sure these boxes are checked:
- [x] Check all the changes in github PR - files count (non of them are redundant, have meaningful changes, all are added), if target branch is correct
- [x] Check methods and variable namings - it should be self descriptive, no typos
- [x] Check you did not introduce breaking changes in API and public models/services.
- [x] Respect extensibility - https://community.virtocommerce.com/t/extensibility-basics-the-domain-model-and-persistence-layer-extension/141
- [x] Follow [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) and [SOLID](https://en.wikipedia.org/wiki/SOLID) principles
- [x] For unit tests - follow Microsoft best practices: https://docs.microsoft.com/en-us/dotnet/core/testing/unit-testing-best-practices
- [x] Consolidate solution dependencies in case you are using newer version, update VC module dependencies in module.manifest. Do not upgrade 3rd party packages that are shipped with the platform with the version newer than in the platform.
- [x] Check code style conventions - https://github.com/VirtoCommerce/styleguide/blob/master/csharp.md
- [x] Check PR have a concise and descriptive title, follow [git commit message rules](https://github.com/VirtoCommerce/styleguide/blob/master/gitcommits.md)

### CODE REVIEW

**Review by AliveMen — CHANGES_REQUESTED** [review]
Please add gif to PR to see how it works now.

**Review by AliveMen — APPROVED** [review]

File: src/VirtoCommerce.Platform.Web/Validators/BlobFolderValidator.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+                .WithMessage(x => $"Minimum length is 3. You entered {x.Name.Length}")
+                .MaximumLength(63)
+                .WithMessage(x => $"Maximum length is 63. You entered {x.Name.Length}")
+                .Must(x => !x.StartsWith("-"))
+                .WithMessage("Folder name must not starts with dash symbol.")
+                .Must(x => !x.EndsWith("-"))
+                .WithMessage("Folder name must not ends with dash symbol.")
+                .Must(x => !x.Contains("--"))
+                .WithMessage("Folder name must not conatin consecutive dash symbols.")
+                .Must(x => new Regex("[0-9a-z -]").IsMatch(x))
```
Comment (tatarincev) [inline src/VirtoCommerce.Platform.Web/Validators/BlobFolderValidator.cs:27]: Seems this regexp will not accept the capital letters.

**Review by tatarincev — COMMENTED** [review]
Could you add the unit tests for validator logic?

**Review by akak1977 — APPROVED** [review]

File: src/VirtoCommerce.Platform.Web/Validators/BlobFolderValidator.cs (hunk 1 above)
Comment (Vectorfield4) [reply src/VirtoCommerce.Platform.Web/Validators/BlobFolderValidator.cs:27]: Rule from BUG: "All letters in a folder name must be lowercase."

**Review by AliveMen — APPROVED** [review]

File: tests/VirtoCommerce.Platform.Tests/Validators/BlobFolderValidatorTests.cs [hunk 2]
```diff
@@ -0,0 +1,158 @@
+using AutoFixture;
+using FluentAssertions;
+using VirtoCommerce.Platform.Core.Assets;
+using VirtoCommerce.Platform.Web.Validators;
+using Xunit;
+
+namespace VirtoCommerce.Platform.Tests.Validators
+{
+    public class BlobFolderValidatorTests : ValidatorTestsHelper
```
Comment (tatarincev) [inline tests/VirtoCommerce.Platform.Tests/Validators/BlobFolderValidatorTests.cs:9]: IMHO all these codes can be simplified by using Data-driven techniques were to one single method is received all test data via [Theory] attributes.

**Review by tatarincev — APPROVED** [review]

**Pull request merged**